tests/unit/models/builder-test.js

```diff
@@ -23,18 +23,101 @@ var MockUI = require('console-ui/mock');
 var Heimdall = require('heimdalljs/heimdall');
 var walkSync = require('walk-sync');
 var itr2Array = require('../../helpers/itr2array');
+var EventEmitter = require('events');
 
 describe('models/builder.js', function() {
   var addon, builder, buildResults, tmpdir;
 
+  function setupBroccoliBuilder() {
+    this.builder = {
+      build: function () {
+        return Promise.resolve('build results');
+      },
+
+      cleanup: function() {
+        return Promise.resolve('cleanup result');
+      }
+    };
+  }
+
+  afterEach(function() {
+    if (builder) {
+      return builder.cleanup();
+    }
+  });
+
+  describe('process signal listeners', function() {
+    var originalListenerCounts;
+
+    function getListenerCount(emitter, event) {
+      if (emitter.listenerCount) { // Present in Node >= 4.0
+        return emitter.listenerCount(event);
+      } else {
+        // deprecated in Node 4.0
+        return EventEmitter.listenerCount(emitter, event);
+      }
+    }
+
+    function getListenerCounts() {
+      return {
+        SIGINT: getListenerCount(process, 'SIGINT'),
+        SIGTERM: getListenerCount(process, 'SIGTERM'),
+        message: getListenerCount(process, 'message'),
+
+        // enable when https://github.com/ember-cli/capture-exit/pull/17 lands
+        // exit: captureExit.listenerCount(),
+      };
+    }
+
+    beforeEach(function() {
+      originalListenerCounts = getListenerCounts();
+    });
+
+    it('sets up listeners for signals', function() {
+      builder = new Builder({
+        setupBroccoliBuilder: setupBroccoliBuilder,
+        project: new MockProject()
+      });
+
+      var actualListeners = getListenerCounts();
+
+      expect(actualListeners).to.eql({
+        SIGINT: originalListenerCounts.SIGINT + 1,
+        SIGTERM: originalListenerCounts.SIGTERM + 1,
+        message: originalListenerCounts.message + 1,
+
+        // enable when https://github.com/ember-cli/capture-exit/pull/17 lands
```

Comment:
I'll create an issue here to make sure we don't forget, but I'd like to land independently (not sure when David/Stef will have time for review, and broccoli-viz has already been broken on master for a few days...).